### PR TITLE
Add exit codes on exit

### LINKED ISSUE #355: Provide exit codes on exit
When we exit we should provide a valid exit code. 

E.g. 0 if there were no problems and something no-zero otherwise.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #355

### CODE REVIEW

File: src/linter.js [hunk 1]
```diff
@@ -438,7 +438,10 @@ export default class Linter {
       })
       .then(() => {
         this.print();
-        return;
+        /* istanbul ignore if  */
```
Comment (muffinresearch) [inline src/linter.js]: This is skipped in the code coverage because the test runs against un-instrumented code.

Comment (tofumatt) [inline src/linter.js]: Makes sense. Can you add this in as an inline comment?

File: tests/test.cli-process.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Tests running the process via the CLI.
```
Comment (tofumatt) [inline tests/test.cli-process.js]: Yay! This is awesome and I was thinking we should be doing this :+1:

**Comment by tofumatt** [comment]
r+wc, :+1:

**Pull request merged**